lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -151,10 +152,13 @@ def parse_index_query_params(
 ) -> HistoryContentsIndexParams:
     """Parses query parameters for the history contents `index` operation
     and returns a model containing the values in the correct type."""
-    return HistoryContentsIndexParams(
-        v=v,
-        dataset_details=parse_dataset_details(dataset_details),
-    )
+    try:
+        return HistoryContentsIndexParams(
+            v=v,
+            dataset_details=parse_dataset_details(dataset_details),
+        )
+    except ValidationError as e:
+        raise validation_error_to_message_exception(e)
```

Comment:
I really think we should get away from this and raise the 422 error. I think the logic in `validation_error_to_message_exception` obfuscates the validation error for discriminating unions.

Reply:
I see, should we just use the default FastAPI's exception handlers instead?

Reply:
I'd prefer this for actual validation errors, but I think we still need this wherever we use legacy endpoints.